### PR TITLE
Cdr messaging

### PR DESCRIPTION
**Pull request by benjag**
Alpha branch with banner refactor and new toast component

### CODE REVIEW

File: src/components/banner/examples/Banner.vue [hunk 1]
```diff
@@ -1,27 +1,149 @@
 <template>
   <div>
     <h2>banner</h2>
+    <cdr-banner type="warning">
```
Comment (mhewson) [inline src/components/banner/examples/Banner.vue:4]: I like the type prop - but I wonder if it is really a type and not a modifier... Applying type="button" or type="input" actually caries application consequence  whereas  type="error" will not even provide something like role="alert"

File: src/components/banner/styles/CdrBanner.scss [hunk 2]
```diff
@@ -3,20 +3,120 @@
...
+    background-color: $cdr-color-background-message-default-02;
+  }
+}
+
+.cdr-banner__message {
+  grid-area: message;
+  padding: $cdr-space-half-x;
+}
+
+.cdr-banner__message-body {
```
Comment (mhewson) [inline src/components/banner/styles/CdrBanner.scss:83]: any reason not to `&-body` and so on here?

File: src/components/banner/styles/CdrBanner.scss [hunk 3]
```diff
@@ -3,20 +3,120 @@
 
 .cdr-banner {
   @include cdr-banner-base-mixin;
+  display: grid;
```
Comment (mhewson) [inline src/components/banner/styles/CdrBanner.scss:6]: I'm a fan of alphabetization for our selector value pairs- open to other organizations though

File: src/components/banner/CdrBanner.jsx [hunk 4]
```diff
@@ -25,14 +25,56 @@ export default {
     baseClass() {
       return 'cdr-banner';
     },
-    iconClass() {
+    borderClass() {
```
Comment (mhewson) [inline src/components/banner/CdrBanner.jsx:28]: do we need all these classes to be unique or can this be set via the parent modifier? 
something like .cdr-banner__info.cdr-banner--border

File: src/components/toast/styles/vars/CdrToast.vars.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+@mixin cdr-toast-base-mixin() {
+  position: relative;
+  border-radius: $cdr-radius-softer;
+  border-left: 4px solid;
+  box-shadow: $cdr-prominence-floating;
+  margin-bottom: $cdr-space-one-x;
+  @include cdr-text-default;
+  @include cdr-sm-mq-up {
+    width: 39rem;
```
Comment (mhewson) [inline src/components/toast/styles/vars/CdrToast.vars.scss:12]: should this be a token?

**Review by mhewson — APPROVED** [review]

File: src/components/banner/styles/CdrBanner.scss (hunk 2 above)
Comment (benjag) [reply src/components/banner/styles/CdrBanner.scss:83]: I'll clean this up

File: src/components/banner/examples/Banner.vue (hunk 1 above)
Comment (benjag) [reply src/components/banner/examples/Banner.vue:4]: Changing would introduce a breaking change to the existing banner API. I kept the same prop in toast for consistency

**Pull request merged**